tests/phpunit/unit-tests/class-llms-test-forms.php

```diff
@@ -384,6 +384,95 @@ public function test_create() {
 
 	}
 
+	/**
+	 * Test forms author on install
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function test_forms_author_on_install() {
+
+		// Clean user* tables.
+		global $wpdb;
+		$wpdb->query( "TRUNCATE TABLE $wpdb->users" );
+		$wpdb->query( "TRUNCATE TABLE $wpdb->usermeta" );
+
+		// Create a subscriber.
+		$subscriber = $this->factory->user->create( array( 'role' => 'subscriber' ) );
+
+		$locs = $this->forms->get_locations();
+
+		// Install forms
+		$installed = $this->forms->install();
+
+		foreach ( $installed as $loc => $id ) {
+			// No admin users, expect 0.
+			$this->assertEquals( 0, get_post( $id )->post_author, $id );
+		}
+
+		// Delete forms.
+		$wpdb->delete( $wpdb->posts, array( 'post_type' => 'llms_form' ) );
+
+		// Create two admins.
+		$admins = $this->factory->user->create_many( 2, array( 'role' => 'administrator' ) );
+
+		// Install forms.
+		$installed = $this->forms->install();
+
+		foreach ( $installed as $loc => $id ) {
+			// Expect the first admin to be the forms author.
+			$this->assertEquals( $admins[0], get_post( $id )->post_author, $id );
+		}
+
+		// Delete forms.
+		$wpdb->delete( $wpdb->posts, array( 'post_type' => 'llms_form' ) );
+
+		// Log in as subscriber.
+		wp_set_current_user( $subscriber );
+
+		// Install forms.
+		$installed = $this->forms->install();
+
+		foreach ( $installed as $loc => $id ) {
+			// Expect the first admin to be the forms author.
+			$this->assertEquals( $admins[0], get_post( $id )->post_author, $id );
+		}
+
+		// Delete forms.
+		$wpdb->delete( $wpdb->posts, array( 'post_type' => 'llms_form' ) );
+
+		// Log in as first admin.
+		wp_set_current_user( $admins[0] );
+
+		// Install forms.
+		$installed = $this->forms->install();
+
+		foreach ( $installed as $loc => $id ) {
+			// Expect the first admin to be the forms author.
+			$this->assertEquals( $admins[0], get_post( $id )->post_author, $id );
+		}
+
+		// Delete forms.
+		$wpdb->delete( $wpdb->posts, array( 'post_type' => 'llms_form' ) );
+
+		// Log in as second admin.
+		wp_set_current_user( $admins[1] );
+
+		// Install forms.
+		$installed = $this->forms->install();
+
+		foreach ( $installed as $loc => $id ) {
+			// Expect the first admin to be the forms author.
+			$this->assertEquals( $admins[1], get_post( $id )->post_author, $id );
+		}
+
+		// Clean user* tables.
+		$wpdb->query( "TRUNCATE TABLE $wpdb->users" );
+		$wpdb->query( "TRUNCATE TABLE $wpdb->usermeta" );
+
```

Comment:
```suggestion
```

We don't need to clean up the test users we created for this test. This won't hurt anything but I'm almost positive it's unnecessary

Reply:
I see what you mean, as of now we wouldn't need that clean up... But I would like to have our tests to not leave anything in the db that could arm another test. If any test would take care of the clean up before quitting (and we generally do that in the `tearDown()`) we wouldn't bore about cleaning at the beginning of a test (when necessary) nor we would spend time on debugging a test that doesn't work as expected (as a result of "environment" changes produced by a previous test).
Am I wrong/missing something obvious/being too much anxious? 😅